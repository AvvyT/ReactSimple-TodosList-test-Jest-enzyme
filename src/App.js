import React, { Component } from 'react';
import './App.css';

let id = 0;

class App extends Component {

  state = { items: [], item: "" }

  changeInput(e) {
    this.setState({ item: e.target.value });
  }

  handleSubmit(e) {
    e.preventDefault();
    id++;

    let newTodo = { name: this.state.item, id: `uniq${id}` };
    let items = [...this.state.items, newTodo];

    this.setState({ items, item: "" });
  }

  handleDelete(id) {
    this.setState({ items: this.state.items.filter(item => id !== item.id) });
  }

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <h1>Todos list</h1>
          <form onSubmit={this.handleSubmit.bind(this)}>
            <input
              className='item'
              placeholder=' Write new item..'
              value={this.state.item}
              onChange={this.changeInput.bind(this)}
            />
            <input
              type='submit'
              value='Add'
            />
          </form>
          <ul>

            {this.state.items.map((item) => (
              <li className='aitem' key={item.id}>
                {item.name}
                <button className="remove"
                  onClick={this.handleDelete.bind(this, item.id)}>x</button>
              </li>
            ))}

          </ul>
        </header>
      </div>
    );
  }
}

export default App;
