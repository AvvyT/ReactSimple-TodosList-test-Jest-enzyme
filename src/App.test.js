import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';

import renderer from 'react-test-renderer';
import { shallow, configure } from 'enzyme';

import Adapter from "enzyme-adapter-react-16"
configure({adapter: new Adapter()});

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

// ----ett system som kallas matchers
it('Olika typer', () => {
  expect('Todo').not.toMatch(/V/);
  const namnLista = ['niklas', 'viktor'];

  expect(namnLista).toContain('niklas');
});

// -------enzyme wrappar
it('Check dom-text', () => {
  const wrapper = shallow(<App />);
  // skapa en mock fun.
  const mockPrevent = jest.fn();

  expect(wrapper.find('li').length).toBe(0)
  // preventDefault-metod: lÃ¤ggs mock-fun
  wrapper.find('form').simulate("submit", {
    preventDefault: mockPrevent
  })
  expect(mockPrevent.mock.calls.length).toBe(1)

  console.log(wrapper.state())
  expect(wrapper.find('li').length).toBe(1)
});

// ------snapshot-testning 
it('renders name correctly', () => {
  const tree = renderer
    .create(<App name={"niklas"} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});