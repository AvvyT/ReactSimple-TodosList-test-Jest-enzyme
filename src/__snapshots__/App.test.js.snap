// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders name correctly 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <h1>
      Todos list
    </h1>
    <form
      onSubmit={[Function]}
    >
      <input
        className="item"
        onChange={[Function]}
        placeholder=" Write new item.."
        value=""
      />
      <input
        type="submit"
        value="Add"
      />
    </form>
    <ul />
  </header>
</div>
`;
